### PR TITLE
OEL-1550: Make oe_content_publication optional in oe_content_sub_entity_document_reference.

### PR DESCRIPTION
**Pull request by escuriola**
### Description

Make oe_content_publication optional in oe_content_sub_entity_document_reference.

### Change log

- Added:
- Changed: oe_publication reference files has been moved from install to optional, remove from info.yml and fixed tests.
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

**Review by drishu — COMMENTED** [review]
I changed the PR target since its part of an epic

File: modules/oe_content_publication/modules/oe_content_publication_document_reference/oe_content_publication_document_reference.info.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+description: Allows to reference publication nodes in document references entities.
+package: OpenEuropa
+type: module
+core_version_requirement: ^9.2
+
+dependencies:
+  - oe_content:oe_content_publication
+  - oe_content:oe_content_sub_entity_document_reference
+
+config_devel:
```
Comment (drishu) [inline modules/oe_content_publication/modules/oe_content_publication_document_reference/oe_content_publication_document_reference.info.yml]: this bit should be removed, but not a blocker

**Review by drishu — CHANGES_REQUESTED** [review]

**Review by drishu — DISMISSED** [review]

**Review by drishu — APPROVED** [review]

File: modules/oe_content_publication/modules/oe_content_publication_document_reference/oe_content_publication_document_reference.info.yml (hunk 1 above)
Comment (brummbar) [reply modules/oe_content_publication/modules/oe_content_publication_document_reference/oe_content_publication_document_reference.info.yml]: Fixed, thanks!

**Review by brummbar — APPROVED** [review]
Approving after @escuriola review.

**Pull request merged**